### ISSUE TITLE
check() does not respect `document = FALSE`

### ISSUE THREAD

**Issue by briandk** [r-lib/devtools#27]
## Expected Behavior

Calling

``` r
check(pkg = "package.name", document = FALSE)
```

should **not** run `document()` on the package being checked.
## Actual Behavior

The `check()` function [currently ignores](https://github.com/hadley/devtools/blob/fa4d4f668c7a7674a48b45a39ada59be7655fa5e/R/check.r#L13-17) the value of `document` passed to it as an argument. Consequently, the `document()`function is always called on `pkg` no matter the input to the function:

``` r
check <- function(pkg = NULL, document = TRUE) {
  pkg <- as.package(pkg)

  document(pkg, clean = TRUE)
  message("Checking ", pkg$package)
```

In short, any time a user calls `check()` on a package that package will be forcibly documented, AND the `document()` function will always flush the roxygen cache, since it's called with `clean = TRUE`
